tests/unit/mathematics/linalg/operations/Eigen3_operations_unittest.cc

```diff
@@ -189,6 +190,35 @@ TEST(LinalgBackendEigen, SGMatrix_add_col_vec_in_place)
 		}
 }
 
+TEST(LinalgBackendEigen, add_diag_vec)
+{
+	const float64_t alpha = 0.7;
```

Comment:
could you do this test more explicit, i.e. just write down the matrices by hand (no loops)
Otherwise good.

Also make sure to use `EXPECT_THROW` for the case of invalid input arugments (the `REQUIRE`)

Finally, test some corner cases (empty matrix doesnt give memory error, etc)